From 21bed213f464b8f2c393fc3fa69030df6772b2b7 Mon Sep 17 00:00:00 2001
From: Andrii Sultanov <andriy.sultanov@vates.tech>
Date: Thu, 24 Apr 2025 09:17:03 +0100
Subject: [PATCH] Drop maybe_daemonize usage, systemd dependency

maybe_daemonize was a noop (since daemon=false by default). Replace
systemd usage with a standalone implementation from Xapi_stdext.

Signed-off-by: Andrii Sultanov <andriy.sultanov@vates.tech>
---
 bin/dune            |  2 +-
 bin/xcp_featured.ml | 16 ++++++----------
 2 files changed, 7 insertions(+), 11 deletions(-)

diff --git a/bin/dune b/bin/dune
index 089e257..129da0b 100644
--- a/bin/dune
+++ b/bin/dune
@@ -11,7 +11,7 @@
  (modes byte exe)
  (names xcp_featured)
  (flags :standard -safe-string)
- (libraries systemd xapi-types xapi-idl.v6))
+ (libraries xapi-types xapi-idl.v6))
 
 (install
  (section bin)
diff --git a/bin/xcp_featured.ml b/bin/xcp_featured.ml
index bead50f..2324ebb 100644
--- a/bin/xcp_featured.ml
+++ b/bin/xcp_featured.ml
@@ -27,17 +27,13 @@ let () =
     ()
   in
 
-  Xcp_service.maybe_daemonize ~start_fn:(fun () ->
-    Debug.set_facility Syslog.Local5;
+  Debug.set_facility Syslog.Local5;
+  Debug.disable "http";
+  handle_shutdown ();
+  Debug.with_thread_associated "main" start server;
 
-    Debug.disable "http";
-
-    handle_shutdown ();
-
-    Debug.with_thread_associated "main" start server;
-
-    ignore (Daemon.notify Daemon.State.Ready)
-  ) ();
+  let module Daemon = Xapi_stdext_unix.Unixext.Daemon in
+  ignore (Daemon.systemd_notify Daemon.State.Ready);
 
   while true do
     Thread.delay 300.;
